PowerDNS 4.7.2 fails to build with gcc13

- Program: Authoritative
 - Issue type: Bug report

### Short description
PowerDNS 4.7.2 fails to build to build with gcc13


```
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.7.2/pdns'
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1       -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -c -o dnsgram.o dnsgram.cc
In file included from statbag.hh:27,
                 from dnsgram.cc:26:
lock.hh: In member function 'T& LockGuardedTryHolder<T>::operator*() const':
lock.hh:235:18: error: 'runtime_error' is not a member of 'std'
  235 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:25:1: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
   24 | #include <shared_mutex>
  +++ |+#include <stdexcept>
   25 | 
lock.hh: In member function 'T* LockGuardedTryHolder<T>::operator->() const':
lock.hh:242:18: error: 'runtime_error' is not a member of 'std'
  242 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:242:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'T& SharedLockGuardedTryHolder<T>::operator*() const':
lock.hh:332:18: error: 'runtime_error' is not a member of 'std'
  332 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:332:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'T* SharedLockGuardedTryHolder<T>::operator->() const':
lock.hh:339:18: error: 'runtime_error' is not a member of 'std'
  339 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:339:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'const T& SharedLockGuardedNonExclusiveTryHolder<T>::operator*() const':
lock.hh:388:18: error: 'runtime_error' is not a member of 'std'
  388 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:388:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'const T* SharedLockGuardedNonExclusiveTryHolder<T>::operator->() const':
lock.hh:395:18: error: 'runtime_error' is not a member of 'std'
  395 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:395:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
make[3]: *** [Makefile:3429: dnsgram.o] Error 1
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.7.2/pdns'
make[3]: *** Waiting for unfinished jobs....
make[3]: Entering directory '/builddir/build/BUILD/pdns-4.7.2/pdns'
```

```
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1       -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -c -o dnsscan.o dnsscan.cc
In file included from statbag.hh:27,
                 from dnsscan.cc:26:
lock.hh: In member function 'T& LockGuardedTryHolder<T>::operator*() const':
lock.hh:235:18: error: 'runtime_error' is not a member of 'std'
  235 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:25:1: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
   24 | #include <shared_mutex>
  +++ |+#include <stdexcept>
   25 | 
lock.hh: In member function 'T* LockGuardedTryHolder<T>::operator->() const':
lock.hh:242:18: error: 'runtime_error' is not a member of 'std'
  242 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:242:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'T& SharedLockGuardedTryHolder<T>::operator*() const':
lock.hh:332:18: error: 'runtime_error' is not a member of 'std'
  332 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:332:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'T* SharedLockGuardedTryHolder<T>::operator->() const':
lock.hh:339:18: error: 'runtime_error' is not a member of 'std'
  339 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:339:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'const T& SharedLockGuardedNonExclusiveTryHolder<T>::operator*() const':
lock.hh:388:18: error: 'runtime_error' is not a member of 'std'
  388 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:388:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'const T* SharedLockGuardedNonExclusiveTryHolder<T>::operator->() const':
lock.hh:395:18: error: 'runtime_error' is not a member of 'std'
  395 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:395:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
make[3]: *** [Makefile:3429: dnsscan.o] Error 1
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.7.2/pdns'
```
```
g++ -std=c++17 -DHAVE_CONFIG_H -I. -I..  -I.. -I.. -pthread  -I../ext/json11 -I../ext/yahttp     -I../ext/protozero/include -DBOOST_CONTAINER_USE_STD_EXCEPTIONS -I/usr/include/p11-kit-1       -DLDAP_DEPRECATED -DSYSCONFDIR=\"/etc/pdns\" -DPKGLIBDIR=\"/usr/lib64/pdns\" -DLOCALSTATEDIR=\"/var/run\" -fPIE -DPIE -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 --param ssp-buffer-size=4 -fstack-protector -g -O2 -Wall -Wextra -Wshadow -Wno-unused-parameter -Wmissing-declarations -Wredundant-decls -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -mbranch-protection=standard -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -c -o dnsreplay.o dnsreplay.cc
In file included from statbag.hh:27,
                 from dnsreplay.cc:66:
lock.hh: In member function 'T& LockGuardedTryHolder<T>::operator*() const':
lock.hh:235:18: error: 'runtime_error' is not a member of 'std'
  235 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:25:1: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
   24 | #include <shared_mutex>
  +++ |+#include <stdexcept>
   25 | 
lock.hh: In member function 'T* LockGuardedTryHolder<T>::operator->() const':
lock.hh:242:18: error: 'runtime_error' is not a member of 'std'
  242 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:242:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'T& SharedLockGuardedTryHolder<T>::operator*() const':
lock.hh:332:18: error: 'runtime_error' is not a member of 'std'
  332 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:332:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'T* SharedLockGuardedTryHolder<T>::operator->() const':
lock.hh:339:18: error: 'runtime_error' is not a member of 'std'
  339 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:339:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'const T& SharedLockGuardedNonExclusiveTryHolder<T>::operator*() const':
lock.hh:388:18: error: 'runtime_error' is not a member of 'std'
  388 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:388:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
lock.hh: In member function 'const T* SharedLockGuardedNonExclusiveTryHolder<T>::operator->() const':
lock.hh:395:18: error: 'runtime_error' is not a member of 'std'
  395 |       throw std::runtime_error("Trying to access data protected by a mutex while the lock has not been acquired");
      |                  ^~~~~~~~~~~~~
lock.hh:395:18: note: 'std::runtime_error' is defined in header '<stdexcept>'; did you forget to '#include <stdexcept>'?
make[3]: *** [Makefile:3429: dnsreplay.o] Error 1
make[3]: Leaving directory '/builddir/build/BUILD/pdns-4.7.2/pdns'
```

Full buildlog is attached.
[build.log](https://github.com/PowerDNS/pdns/files/10480048/build.log)

--- a/pdns/lock.hh
+++ b/pdns/lock.hh
@@ -22,6 +22,7 @@
 #pragma once
 #include <mutex>
 #include <shared_mutex>
+#include <stdexcept>
 
 /*
   This file provides several features around locks:

